### PR TITLE
Lizardconomy

### PR DESCRIPTION
**Pull request by coiax**
FantasticFwoosh says:

> Within the ash walker compound is a vendor, it is essentially just a front door of a steel locker attached to a upright cairn with some beady yellow eyes beaming through a little slit entrance. Its not a machine that draws power but can be dragged away for amusing station usage.
> 
> Inside 'aesthetically speaking' is a merchant who has shop UI to buy specific sets of things for coins. The base would also require a ore redemption machine and a coin presser. (meaning that aside from some small time loot, most of the crates junking it up can be thrown out)
> 
> Basically as a sub-objective of the ash walkers existance, they can go around picking up minerals, returning and processing them (for obvious benefits of having materials to work with, such as crafting bolas & such) and then coin press the end results to pay off the vendor for supplies. Not quite golem'esque R&D but adds depth. Adding unique items to the ash walker vendor may also spice things up, for a virtually unlimited supply of pickaxes, mining bags, spears and other trophies/useful objects.
> 
> Money purses could also be a thing for containing personal lizard wealth. Think lizard game of thrones.
- [ ] make the "ash shack", the mysterious shopkeeper shack where ash lizards can bargain with a mysterious shopkeeper lizard
- [ ] make ash walkers able to get coins, somehow?
- [ ] make wallets able to store, say, up to 20 coins
- [ ] craftable coin hoards, made out of coins, and deconstructable to get the coins back

### CODE REVIEW

**Comment by KorPhaeron** [comment]
Not sure how I feel about this. Group mining for a vendor is already what golems do, where as the lizards are far more disposable/combat oriented.

I'd rather keep them hunter/gatherer, though I realize there is little to gather right now. Someone sent me a good design document for ash lizards which I will repost when I am back at my computer.

tl;dr I would prefer lizard currency largely remain flesh.

That being said you guys made golems way better so maybe I should just see how this plays out

**Comment by FantasticFwoosh** [comment]
One of the slight issues of ashwalker is that combat not only does have some cost attached (medkits/supplies of spears/tools etc) but actively and aggressively forces combat upon the miners as a primary means to being with not much else other than making eggs. The RP aspect is the enrichment of that culture, and i respect your opinions Kor, its one of the defining features that divides methodical playstyles of golems from the earthy RP approach & interpret everything and poke it with a spear which is engrossing.

By expanding the depth of ashwalkers they can support larger numbers of players at a time doing seperate jobs (highpop in mind, where supplies are thin) and sustainably cover rising costs of more and more ambitious hunts on tendrils and the station, as well as allow basic artistic freedom for different tastes in which to express some RP.

Under recent coin-cargo merges, the station would be incentivised to actively force further conflict as-well for much needed credits also, much akin to Christopher Columbus and the people of south America (that's actually a very good comparison, and a excellent plan for killing off ash-walkers with smallpox)

**Comment by Zilenan91** [comment]
How about making their currency cups of gibs and various organs

**Comment by FantasticFwoosh** [comment]
We have dinnerware supplies in the autolathe, it'd require one or a absolute buttload in a crate. All in all not the most practical of things (throwing gibs everywhere accidentally or abusing instant re-spawning by stripping and committing suicide)

Its kinda silly to offer the ash-shack (see commit and original idea thread) a glass of guts when the owner literally watches his previous customers explode in the gory glory of the necropolis and could reach over and say its fine i've been paid.

**Comment by Zilenan91** [comment]
You can scoop up guts with beakers, so they'd just need a crate of beakers/drinking glasses

**Comment by FantasticFwoosh** [comment]
Lizard people supplies are very loose and prone to getting lost in lava or just lost (which is why the ash-shack is in this PR to sell supplies) - Also see the part where lizards literally powergame suicide rush into tendril for free guts, it'd only get abused, and literally every corpse they bring back would inflate the static economy to the point it doesn't even matter (equalling not fun or balanced).

Coins are a manifestible thing of value that can be made. A ash walker who comes across a greed ruin is a very happy lizard indeed but chances with map generation of having two at the same time in close proximity is very slim.

**Comment by Zilenan91** [comment]
You do realize it takes three corpses to spawn an Ash Walker right? So them suiciding into the tendril wouldn't do a whole lot.

**Comment by FantasticFwoosh** [comment]
Still counts. Plus the ash walkers are always hunting and can count to make up the numbers, literally 2 hunts + 1 suicide equals instant re-spawn rinse and repeat ding dong scoop those gibs up and get rich.

**Comment by Zilenan91** [comment]
...Then price everything very highly?

**Comment by FantasticFwoosh** [comment]
Lizards will literally powergame to cycle gibs. It wouldnt be worth it.

Coins are honestly the most balanced at the moment. Double incentivised to kill watchers for diamond ores in order to maintain the economy or otherwise pick up loose minerals and dig tunnels around which helps lizard exploration in and around the starting area and stretching further out.

**Comment by coiax** [comment]
If we're going with a coin economy, then you should be able to sell goliath meat (raw and cooked) and plates at the ash shack for coins. Possibly the same for diamond ore. Then they can start acrewing wealth without having to have a full smeltery/mint setup.

**Comment by RemieRichards** [comment]
@Zilenan91 no it's 2 corpses.

**Comment by FantasticFwoosh** [comment]
Manufacturing coins would be more profitable but risks being inflated for higher value minerals. Ideally if ash walkers have a lot of rubbish metal coins, cargo is probably going to leave them alone compared to ash walkers who prioritise taking more valuable gold/silver/diamond coins to pay for better equipment.

Why would you straight up sell diamond ore and monster drops (debatable) when you can buy fuel tanks to smelt it into diamonds with a welder (/some kind of furnace or ORM) and push it into a press for 5 coins? (unless coin pressers get nerfed or ash walkers get a basic 2-3 coin per ore presser)

**Comment by FantasticFwoosh** [comment]
#16934 

We have viable money bag refactors now. Unfortunately no coin stacking, but space cash can be collected.

**Comment by KorPhaeron** [comment]
Doesn't seem to be any code yet after a week, re-open this when there is stuff to review/debate about
